integrations/operator/controllers/resources/base_reconciler.go

```diff
@@ -79,15 +88,24 @@ func (r ResourceBaseReconciler) Do(ctx context.Context, req ctrl.Request, obj kc
 		return ctrl.Result{}, trace.Wrap(err)
 	}
 
+	if checkAnnotationFlag(obj, AnnotationFlagIgnore, false /* defaults to false */) {
+		log.Info(fmt.Sprintf("Resource is flagged with annotation %q, it will not be reconciled.", AnnotationFlagIgnore))
+		return ctrl.Result{}, nil
+	}
+
 	hasDeletionFinalizer := controllerutil.ContainsFinalizer(obj, DeletionFinalizer)
 	isMarkedToBeDeleted := !obj.GetDeletionTimestamp().IsZero()
 
 	// Delete
 	if isMarkedToBeDeleted {
 		if hasDeletionFinalizer {
-			log.Info("deleting object in Teleport")
-			if err := r.DeleteExternal(ctx, obj); err != nil && !trace.IsNotFound(err) {
-				return ctrl.Result{}, trace.Wrap(err)
+			if checkAnnotationFlag(obj, AnnotationFlagKeep, false /* defaults to false */) {
```

Comment:
I would create specific method for each annotation flags instead of adding argument flags.

```go
if hasAnnotationKeepFlag(obj) {
    //skip
}